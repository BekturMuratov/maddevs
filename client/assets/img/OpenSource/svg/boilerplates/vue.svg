<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>