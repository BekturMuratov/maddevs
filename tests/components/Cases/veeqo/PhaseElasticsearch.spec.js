import PhaseElasticsearch from '@/components/Cases/veeqo/PhaseElasticsearch'
import { render } from '@testing-library/vue'

describe('PhaseElasticsearch component', () => {
  it('should render correctly', () => {
    const { container } = render(PhaseElasticsearch)

    expect(container).toMatchSnapshot()
  })
})
