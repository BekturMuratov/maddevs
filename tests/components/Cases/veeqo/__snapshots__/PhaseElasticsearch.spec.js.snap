// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhaseElasticsearch component should render correctly 1`] = `
<div>
  <section
    class="container_regular m-66_bottom media-m-48_bottom"
  >
    <h2
      class="case_title_h2 m-12_bottom"
    >
      
    Phase 3: Elasticsearch
  
    </h2>
     
    <p
      class="case_paragraph m-24_bottom"
    >
      
    Elasticsearch is crucial in the way users experience the Veeqo platform: the dashboard and the entire interface rely on Elasticsearch. Even if everything else functions flawlessly, delays in the search engine alone cause problems for user experience.
  
    </p>
     
    <div
      class="m-96_bottom media-m-48_bottom"
    >
      <div
        class="case_card background-color-silver"
      >
        <h3
          class="case_title_h4 m-8_bottom"
        >
          
        Challenge
      
        </h3>
         
        <p
          class="case_paragraph"
        >
          
        Elasticsearch didnâ€™t cope with the load due to its outdated version and non-optimized configurations.
      
        </p>
      </div>
       
      <div
        class="case_card background-color-blue-light"
      >
        <h3
          class="case_title_h4 m-8_bottom"
        >
          
        Effect on users
      
        </h3>
         
        <p
          class="case_paragraph"
        >
          
        The users were often unable to access search results. Elasticsearch would expose them to delays of up to 30 seconds.
      
        </p>
      </div>
    </div>
     
    <h3
      class="case_title_h4 m-8_bottom"
    >
      
    Solution
  
    </h3>
     
    <p
      class="case_paragraph m-24_bottom media-m-12_bottom"
    >
      
    There are two main ways to improve performance: Increase compute resources, Optimize the use of resources
  
    </p>
     
    <p
      class="case_paragraph m-34_bottom media-m-24_bottom"
    >
      
    At any given moment DevOps specialists calculate and evaluate cost factors of different solutions. We applied both types of solutions as we started with enhancing the cluster and later optimized indexing to make searching as convenient for users as possible.
  
    </p>
     
    <div>
      <div
        class="case_card background-color-silver"
      >
        <h4
          class="case_title_h4 m-8_bottom"
        >
          
        New cluster
      
        </h4>
         
        <p
          class="case_paragraph m-24_bottom media-m-12_bottom"
        >
          
        We implemented a new Elasticsearch cluster as a self-hosted solution on EC2 instances.
      
        </p>
         
        <p
          class="case_paragraph case_bold"
        >
          
        Three reasons:
      
        </p>
         
        <ul
          class="case_list"
        >
          <li
            class="list-item_dot"
          >
            Control over the security of our Elasticsearch
          </li>
           
          <li
            class="list-item_dot"
          >
            Immediate response in case of incidents
          </li>
           
          <li
            class="list-item_dot"
          >
            Independent monitoring and investigation of Elasticsearch performance
          </li>
        </ul>
      </div>
       
      <div
        class="case_card background-color-blue-light"
      >
        <h4
          class="case_title_h4 m-8_bottom"
        >
          
        Indexing and performance
      
        </h4>
         
        <p
          class="case_paragraph m-24_bottom media-m-12_bottom"
        >
          
        The indices featured too much unnecessary information and non-optimized mapping.
      
        </p>
         
        <p
          class="case_paragraph case_bold"
        >
          
        What we did:
      
        </p>
         
        <ul
          class="case_list"
        >
          <li
            class="list-item_dot"
          >
            Refactored them to make them concise
          </li>
           
          <li
            class="list-item_dot"
          >
            Used the replication and sharding mechanisms to distribute the indices among five nodes
          </li>
        </ul>
      </div>
    </div>
  </section>
</div>
`;
